src/Workspaces/Core/Portable/NamingStyles/Serialization/NamingStylePreferences.cs

```diff
@@ -339,5 +336,36 @@ public override int GetHashCode()
   </NamingRules>
 </NamingPreferencesInfo>
 ";
+
+        private static XElement GetUpgradedSerializationIfNecessary(XElement rootElement)
+        {
+            var serializationVersion = int.Parse(rootElement.Attribute("SerializationVersion").Value);
+
+            if (serializationVersion == 4)
+            {
+                UpgradeSerialization_4To5(rootElement = new XElement(rootElement));
+                serializationVersion = 5;
+            }
+
+            // Add future version checks here. If the version is off by more than 1, these upgrades will run in sequence.
+            // The next one should check serializationVersion == 5 and update it to 6.
+            Debug.Assert(s_serializationVersion == 5, "After increasing the serialization version, add an upgrade path here.");
+
+            return serializationVersion == s_serializationVersion
+                ? rootElement
+                : XElement.Parse(DefaultNamingPreferencesString);
+        }
+
+        private static void UpgradeSerialization_4To5(XElement rootElement)
+        {
+            var methodElements = rootElement
+                .Descendants()
+                .Where(e => e.Name.LocalName == "SymbolKind" && e.Value == "Method").ToList();
+
+            foreach (var element in methodElements)
+            {
+                element.ReplaceWith(XElement.Parse("<MethodKind>Ordinary</MethodKind>"));
```

Comment:
should we replace, or just augment?

Reply:
Doesn't matter if we augment... when saving it would ignore the other option and only store this.